### PR TITLE
WIP [Sleet] Pass Google Pay token in Adyen Auth Request

### PR DESCRIPTION
**Pull request by anooj-bolt**
Pass Google Pay token in Adyen Gateway Auth Request.

As documented here: https://docs.adyen.com/payment-methods/google-pay/api-only#before-you-begin.

Ticket: https://app.asana.com/0/1200616417881435/1203366263383020/f

### CODE REVIEW

File: gateways/adyen/request_builder.go [hunk 1]
```diff
@@ -92,7 +93,12 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest, merchantAccount s
 	}
 
 	level3 := authRequest.Level3Data
-	if level3 != nil {
+	if authRequest.Options[googlePayTokenOption] != nil {
+		googlePayData := map[string]interface{}{
+			"payment.token": authRequest.Options[googlePayTokenOption].(string),
+		}
+		request.AdditionalData = googlePayData
+	} else if level3 != nil {
```
Comment (rlcooper46) [inline gateways/adyen/request_builder.go:101]: I think with this logic, we will skip building the level3 data if google pay is present. Lets adjust to do both

Comment (anooj-bolt) [reply gateways/adyen/request_builder.go:101]: We don't want to build the l3 data for Google Pay

Comment (rlcooper46) [reply gateways/adyen/request_builder.go:101]: we should send both, its up to adyen to decide if the l2l3 data applies to the transaction

Comment (anooj-bolt) [reply gateways/adyen/request_builder.go:106]: Addressed

**Review by yuweihsu01 — COMMENTED** [review]
I think for Adyen Gateway, it used Checkout API and you should follow https://docs.adyen.com/payment-methods/google-pay/api-only#before-you-begin.

**Comment by yuweihsu01** [comment]
And it seems we are using v65 [[DD](https://app.datadoghq.com/logs?query=%22live.adyenpayments.com%2Fcheckout%2F%22&cols=service%2C%40request_path&event=AQAAAYSHdIHY2TdP_gAAAABBWVNIZEpHMUFBQ0xmU1pzYzZEM3l3QUU&index=main&messageDisplay=inline&stream_sort=time%2Cdesc&viz=stream&from_ts=1668630682826&to_ts=1668717082826&live=true)], but the requirement seems to be v67. Have asked them in email.

File: gateways/adyen/request_builder.go [hunk 2]
```diff
@@ -92,7 +93,17 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest, merchantAccount s
...
+		googlePayData := map[string]interface{}{
+			"payment.token": authRequest.Options[googlePayTokenOption].(string),
+		}
+		request.AdditionalData = googlePayData
+	} else if level3 != nil && authRequest.Options[googlePayTokenOption] != nil {
+		var builtL3Data map[string]string
+		builtL3Data = buildLevel3Data(level3)
+		builtL3Data["payment.token"] = authRequest.Options[googlePayTokenOption].(string)
+		request.AdditionalData = builtL3Data
+	} else if level3 != nil {
```
Comment (rlcooper46) [inline gateways/adyen/request_builder.go:106]: nit: we could probably simplify this a bit?
```suggestion
	if authRequest.Options[googlePayTokenOption] != nil {
		googlePayData := map[string]interface{}{
			"payment.token": authRequest.Options[googlePayTokenOption].(string),
		}
		request.AdditionalData = googlePayData
	} 
	if level3 != nil  {
	        if request.AdditionalData == nil {
	            request.AdditionalData = make(map[string]string)
	        }
		buildLevel3Data(level3, request.AdditionalData)
	} 
```

Comment (anooj-bolt) [reply gateways/adyen/request_builder.go:106]: actually getting rid of this as its the wrong approach

**Review by yuweihsu01 — APPROVED** [review]
LGTM

**Pull request merged**